apps/api/src/app/layouts/use-cases/create-layout/create-layout.use-case.ts

```diff
@@ -4,14 +4,21 @@ import { Injectable } from '@nestjs/common';
 import { CreateLayoutCommand } from './create-layout.command';
 
 import { LayoutDto } from '../../dtos/layout.dto';
-import { ChannelTypeEnum, IEmailBlock } from '../../types';
+import { ChannelTypeEnum, ITemplateVariable } from '../../types';
+import { ContentService } from '../../../shared/helpers/content.service';
 
 @Injectable()
 export class CreateLayoutUseCase {
   constructor(private layoutRepository: LayoutRepository) {}
 
   async execute(command: CreateLayoutCommand) {
-    const entity = this.mapToEntity(command);
+    const contentService = new ContentService();
+    const extractedVariables = contentService.extractVariables(command.content);
+    const variables = [
+      ...new Map([...extractedVariables, ...(command.variables || [])].map((item) => [item.name, item])).values(),
+    ] as ITemplateVariable[];
```

Comment:
Eeef, yuck. I will refactor this. Just wanted to see that it works as it should end to end. 

What I want to do here is first extract the variables ourselves (as we talked about, we should calculate variables in BE for the external use of this endpoint) But. I still want to allow the variable fields setting in the layout editor - like default value and required. 

So take extractedVariables but take defaultValue out of command.variables  i